soap/soap.go

```diff
@@ -211,6 +223,10 @@ type Client struct {
 	headers []interface{}
 }
 
+type httpClient interface {
```

Comment:
Usually you want to expose public interface as WithHTTPClient accepts it and consuming client may want to use it also.

Reply:
I don't think users of this library should be using the interface as a return type as it violates the proverb "accept interfaces but return structs" which is exactly this interface is doing. That said, I don't care too much either way.